snuba/datasets/generic_metrics.py

```diff
@@ -0,0 +1,16 @@
+from typing import Sequence
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.entities import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.entity import Entity
+
+DEFAULT_GRANULARITY = 60
+
+
+class GenericMetricsDataset(Dataset):
```

Comment:
yeah, I was having trouble getting that working but I'll try again, I agree that makes sense

Reply:
yeah, I don't think this can work with e2e testing without major changes. The application is initialized in `BaseApiTest`, and at that point `settings.ENABLE_DEV_FEATURES` is already False (so we get a test error for the `generic_metrics` dataset not being defined). I tried setting it to true before the `import application` line here: https://github.com/getsentry/snuba/blob/8a31d3066810d5ed55276dbfb16c008a1720f17f/tests/base.py#L6

but I think something else may have already imported the dataset factory elsewhere and `generic_metrics` is still not set in [DATASET_NAMES](https://github.com/getsentry/snuba/blob/8a31d3066810d5ed55276dbfb16c008a1720f17f/snuba/datasets/factory.py#L13)